src/arraydist.jl

```diff
@@ -3,7 +3,7 @@
 const VectorOfUnivariate = Distributions.Product
 
 function arraydist(dists::AbstractVector{<:UnivariateDistribution})
-    return Product(dists)
+    return product_distribution(dists)
```

Comment:
This changes the return type and breaks tests, as observed in https://github.com/TuringLang/DistributionsAD.jl/pull/228#discussion_r928239304.

Reply:
Let's see whether #228 fixes broken tests.